completion/available/svn.completion.bash

```diff
@@ -1,1514 +1,40 @@
-		-*)     optBase=${opt:0:2} ;;
-		esac
-
-		cmdOpts=" $cmdOpts "
-		cmdOpts=${cmdOpts/ ${optBase} / }
-
-		# take out alternatives
-		case $optBase in
-		-h)              cmdOpts=${cmdOpts/ --help / } ;;
-		--help)          cmdOpts=${cmdOpts/ -h / } ;;
-		-q)              cmdOpts=${cmdOpts/ --quiet / } ;;
-		--quiet)         cmdOpts=${cmdOpts/ -q / } ;;
-		esac
-
-		# skip next option if this one requires a parameter
-		if [[ $opt == @($optsParam) ]] ; then
-			((++i))
-		fi
-	done
-
-	COMPREPLY=( $( compgen -W "$cmdOpts" -- $cur ) )
-
-	return 0
-}
-complete -F _svnsync -o default svnsync
-
-# reasonable completion for 'svnversion'
-_svnversion ()
-{
-	local cmdOpts=" -n --no-newline -c --committed -h --help --version "
-	local cur=${COMP_WORDS[COMP_CWORD]}
-
-	COMPREPLY=()
-
-	# parse current options
-	local options= wcpath= trailurl= last='none' stat= opt= i=-1 isCur=
-	for opt in ${COMP_WORDS[@]}
-	do
-		[[ $i -eq $COMP_CWORD ]] && stat=$last
-		let i++
-
-		# are we processing the current word?
-		isCur=
-		[[ $i -eq $COMP_CWORD ]] && isCur=1
-
-		# skip first command, should be 'svnversion'
-		if [ $last = 'none' ] ; then
-			last='first'
-			continue
-		fi
-
-		# get options
-		if [[ $last != 'arg' && $opt == -* ]]
-		then
-			# if '--' is at the current position, it means that we are looking
-			# for '--*' options, and not the end of option processing.
-			if [[ $opt = '--' && ! $isCur ]]
-			then
-				last='arg'
-			else
-				options="$options $opt "
-				last='opt'
-			fi
-			continue
-		fi
-		# get arguments
-		if [[ $opt != -* ]]
-		then
-			last='arg'
-			if [[ ! $wcpath ]]
-			then
-				wcpath=$opt
-			elif [[ ! $trailurl ]]
-			then
-				trailurl=$opt
-			fi
-		fi
-	done
-	[[ $stat ]] || stat=$last
-
-	# argument part
-	if [[ $cur != -* || $stat = 'arg' ]]
-	then
-		[[ $wcpath && $trailurl ]] && COMPREPLY=( '' )
-		return 0
-	fi
-
-	# suggest options, and  take out already given options
-	for opt in $options
-	do
-		# take out options
-		cmdOpts=${cmdOpts/ $opt / }
-
-		# take out alternatives
-		case $opt in
-			-n)              cmdOpts=${cmdOpts/ --no-newline / } ;;
-			--no-newline)    cmdOpts=${cmdOpts/ -n / } ;;
-			-h)              cmdOpts=${cmdOpts/ --help / } ;;
-			--help)          cmdOpts=${cmdOpts/ -h / } ;;
-			-c)              cmdOpts=${cmdOpts/ --committed / } ;;
-			--committed)     cmdOpts=${cmdOpts/ -c / } ;;
-		esac
-	done
+# Locate and load completions for `svn`.
 
-	COMPREPLY=( $( compgen -W "$cmdOpts" -- $cur ) )
+# Make sure svn is installed
+_command_exists svn || return
 
+# Don't handle completion if it's already managed
+if _completion_exists svn; then
+	_log_warning "completion already loaded - this usually means it is safe to stop using this completion"
 	return 0
-}
-# -X option does not seem to work?
-complete -F _svnversion -o dirnames -X '*.svn*' svnversion
+fi
+
+_svn_bash_completion_xcrun_svn=
+if _command_exists xcrun; then
+	_svn_bash_completion_xcrun_svn="$(xcrun --find svn)"
```

Comment:
On Mac, if any Xcode tools (app or "Command Line Tools") are installed, then `xcrun` will return the path of the real binary instead of the stub. Only `svn` on one version of OSX has a stub without a real actual binary, since Xcode was updated to remove subversion but that version of Xcode was back-ported to run on the previous release of OSX and the root partition is read-only so the Xcode installer can't remove the stub.

Reply:
That was quite informative, thank you - I'm still a bit uneasy in just blindly using the result of `xcrun` without checking the error code of the command.

Reply:
I can add a check on the error code, but if it returns a valid file path...then that's good enough IMO... If what it returns isn't a valid file path, then we fail anyway no matter what the return code.